resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/rest/kubernetes/v2/ResourceStagingServerSslOptionsProviderSuite.scala

```diff
@@ -87,6 +87,32 @@ class ResourceStagingServerSslOptionsProviderSuite extends SparkFunSuite with Be
     }
   }
 
+  test("Setting pem files without setting passwords should use random passwords.") {
```

Comment:
I think we want at least one integration test that verifies the whole flow works with full-on SSL/certs everywhere, including this PR to use client certificates from init container -> resource staging server.  Is there an existing test we can extend to have more SSL, or does that need to be a new test to get that coverage?